solr/core/src/test/org/apache/solr/handler/RequestHandlerMetricsTest.java

```diff
@@ -0,0 +1,196 @@
+import org.apache.solr.common.SolrInputDocument;
+import org.apache.solr.common.params.SolrParams;
+import org.apache.solr.common.util.NamedList;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class RequestHandlerMetricsTest extends SolrCloudTestCase {
+
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    System.setProperty("metricsEnabled", "true");
+    configureCluster(1)
+        .addConfig(
+            "conf1",
+            TEST_PATH()
+                .resolve("configsets")
+                .resolve("cloud-aggregate-node-metrics")
+                .resolve("conf"))
+        .configure();
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    super.setUp();
+    cluster.deleteAllCollections();
+  }
+
+  @AfterClass
+  public static void afterClass() {
+    System.clearProperty("metricsEnabled");
+  }
+
+  @Test
+  @SuppressWarnings({"unchecked"})
+  public void testAggregateNodeLevelMetrics() throws SolrServerException, IOException {
+    String collection1 = "testRequestHandlerMetrics1";
+    String collection2 = "testRequestHandlerMetrics2";
+
+    CloudSolrClient cloudClient = cluster.getSolrClient();
+
+    CollectionAdminRequest.Create create =
+        CollectionAdminRequest.createCollection(collection1, "conf1", 1, 1);
+    cloudClient.request(create);
+    cluster.waitForActiveCollection(collection1, 1, 1);
+
+    create = CollectionAdminRequest.createCollection(collection2, "conf1", 1, 1);
+    cloudClient.request(create);
+    cluster.waitForActiveCollection(collection2, 1, 1);
+
+    SolrInputDocument solrInputDocument =
+        new SolrInputDocument("id", "10", "title", "test", "val_s1", "aaa");
+    cloudClient.add(collection1, solrInputDocument);
+    cloudClient.add(collection2, solrInputDocument);
+
+    SolrQuery solrQuery = new SolrQuery("*:*");
+    cloudClient.query(collection1, solrQuery);
+    cloudClient.query(collection2, solrQuery);
+
+    NamedList<Object> response =
+        cloudClient.request(
+            new SolrRequest<>(SolrRequest.METHOD.GET, "/admin/metrics") {
+              @Override
+              public String getRequestType() {
+                return "metricsRequest";
+              }
+
+              @Override
+              public SolrParams getParams() {
+                return null;
+              }
+
+              @Override
+              protected SolrResponse createResponse(SolrClient client) {
+                return new SolrResponseBase();
+              }
+            });
+
+    NamedList<Object> metrics = (NamedList<Object>) response.get("metrics");
+
+    final double[] minQueryTime = {Double.MAX_VALUE};
+    final double[] maxQueryTime = {-1.0};
+    final double[] minUpdateTime = {Double.MAX_VALUE};
+    final double[] maxUpdateTime = {-1.0};
```

Comment:
Why are these arrays? I am probably missing something fairly straightforward here.

Reply:
They have to be final or effectively final to be used in the lambda function below so it is a common pattern to use an array which can be final and allow the value to be updated. Could also use an AtomicReference, but doesn't seem to make a difference in this test.

Reply:
ah, I missed that it was in side of a lambda! thanks!